### ISSUE TITLE
Range without steps gives NaN

### ISSUE THREAD

**Issue by MartijnRasenberg** [leongersen/noUiSlider#948]
I'm using the noUISlider for a planning system. It works great with the exception of one problem. When I have a range defined (between other ranges) with no step it causes an error and the handler bar hangs eventually.

The functional setup is like this:
![image](https://user-images.githubusercontent.com/490467/52626725-17cfdc00-2eb4-11e9-923f-9a243706db25.png)
This is a week planning of doors where there might be a demand of 0 in some weeks. The more the handler is slided to the right, the more doors are to be produced. 

See https://codepen.io/TambienNL/pen/daKoep for a code example. Drag the slider around and let it stop at the left border of a range without step. Usually the first time you try to pickup the slider and move it around after this works, the 2nd and more times it fails and the handlerbar stays in place.

It seems to me that this function
function fromPercentage(range, value)
is part of the problem. This function returns IsNaN when the handler is hovered over a range with no step. This is caused by _(range[1] - range[0]) = 0_ and divide by zero gives the problem. If I fix that in this function the handler bar never hangs anymore, but the range with no step has a huge number of steps when dragging. So that is just a partial fix.

Would a permanent fix be possible?

**Comment by leongersen** [comment]
I can confirm this doesn't work as intended. Looking into it.

**Comment by leongersen** [comment]
In your example, the issue is in:

```JavaScript
        '16.666666666666668%':[1239, 1],
        '33.333333333333336%':[1239, 1],
```

I'm not sure if you intended to have a zero-length range?

Either way, the slider shouldn't crash on this configuration.

The fix would be (on line 336):

```JavaScript
        // Ignore 'false' stepping and 0-length ranges.
        if (!n || that.xVal[i] === that.xVal[i + 1]) {
            return true;
        }
```

**Comment by MartijnRasenberg** [comment]
Hi Leon

Thanks for a quick answer!
Indeed, that was the intention and where the problem occurs.

The fix indeed does prevent the NaN result but it leaves the zero-length range with multiple steps. The intent was that the zero-length range would have zero steps as well, the slider should "skip" the range. Is that an option?

**Comment by leongersen** [comment]
Could you elaborate on why you need the sub range at `33.333`? Removing that line would give the correct output, yes?

**Comment by MartijnRasenberg** [comment]
check the screenshot above. The intent is to "add" a forecast to a planning.
In this case for example:
![image](https://user-images.githubusercontent.com/490467/52639230-16161080-2ed4-11e9-8587-dbd731f777fe.png)
Every column is a week and in the cell's are the forecast (sold items) for that week. The slider determines how many items will be produced by sliding the slider right / left. The sum of the value cells left of the handler (shown in green) is planned / show in the input on the left. Each cell has steps for the value in the cell. So a cell with 4 has 4 steps, with 3 has 3 and with 1239 has 1239.
There are weeks (columns) with no items sold, the ones with 0 in it. The slider should skip that week (no steps). This actually works fine, but returns a NaN causing the slider to hang.

**Comment by leongersen** [comment]
It took me a couple of reads, but I understand why you are doing what you are doing.

I see two options: adding an exception if there are any zero-length ranges in the data (this would not help you), or setting the number of steps in a zero-length range to 1. I'll see if the latter is doable.

**Comment by MartijnRasenberg** [comment]
> I'll see if the latter is doable.

That would be great, dank je!

**Comment by leongersen** [comment]
It turned out to be pretty easy. Released in noUiSlider 13.1.1. Thanks for the clear bug report!
